frontend/src/pages/instance/Snapshots/index.vue

```diff
@@ -259,13 +256,15 @@ export default {
             this.$emit('instance-updated')
         },
         async downloadSnapshotPackage (snapshot) {
-            const ss = await SnapshotApi.getSnapshot(this.instance.id, snapshot.id)
+            const ss = await SnapshotsApi.getFullSnapshot(snapshot.id)
+            const owner = snapshot.device || snapshot.project
+            const ownerType = snapshot.device ? 'device' : 'instance'
             const packageJSON = {
-                name: this.instance.safeName,
-                description: `${snapshot.name} - ${snapshot.description}`,
+                name: `${owner.safeName || owner.name}`.replace(/[^a-zA-Z0-9-]/g, '-').toLowerCase(),
```

Comment:
The `.replace(/[^a-zA-Z0-9-]/g, '-').toLowerCase()` could be extracted as a separate composable into the string composable `frontend/src/composables/String.js:5` and reused

Reply:
not in scope at this time (as discussed)